python/lsst/ts/ess/common/sensor/boltek_ld250.py

```diff
@@ -0,0 +1,148 @@
+
+    Noise data is output by the instrument if the noise threshold is exceeded.
+    The threshold can be configured in the detector and is set such that the
+    noise data never, or at least as little as possible, is emitted. Still, it
+    is possible that such a message is emitted so this sensor class needs to be
+    able to process it. The format is:
+
+        '$WIMLN*<cs><\r><\n>'
+
+    where:
+
+        $           Start of telemetry indicator.
+        WIMLN       Telemetry prefix.
+        cs          Hex checksum (Range 00 to FF).
+        <\r><\n>    2-character terminator.
+
+    Strike data is output by the instrument once per minute in case one or more
+    strikes are detected. The format is:
+
+        '$WIMLI,<ddd>,<uuu>,<bbb.b>*<cs><\r><\n>'
+
+    where:
+
+        $           Start of telemetry indicator.
+        WIMLI       Telemetry prefix.
+        ddd         Corrected strike distance (Range 0 to 300 miles).
+        uuu         Uncorrected strike distance (Range 0 to 300 miles).
+        bbb.b       Bearing to the strike (Range 000.0 to 359.9 deg azimuth).
+        cs          Hex checksum (Range 00 to FF).
+        <\r><\n>    2-character terminator.
+
+    The placeholders shown for the values are displaying a non-zero-padded
+    value for int and a zero-padded value for float values.
+
+    Parameters
+    ----------
+    log : `logger`
+        The logger for which to create a child logger.
+    """
+
+    def __init__(
+        self,
+        log: logging.Logger,
+    ) -> None:
+        super().__init__(log=log)
+
+    async def extract_telemetry(self, line: str) -> list[float | int | str]:
+        # stripped_line: str = line.strip(self.terminator)
+        # line_items = stripped_line.split(self.delimiter)
+        output: list[str | int | float] = []
+
+        # Note that group(0) matches the whole pattern so that needs to be
+        # skipped whenever groups in the match are accessed.
+        status_match = STATUS_PATTERN.match(line)
+        if status_match:
+            output.append(status_match.group(1))
+            output.append(int(status_match.group(2)))
+            output.append(int(status_match.group(3)))
+            output.append(int(status_match.group(4)))
+            output.append(int(status_match.group(5)))
+            output.append(float(status_match.group(6)))
+        else:
+            strike_match = STRIKE_PATTERN.match(line)
+            if strike_match:
+                output.append(strike_match.group(1))
+                output.append(int(strike_match.group(2)))
+                output.append(int(strike_match.group(3)))
+                output.append(float(strike_match.group(4)))
+            else:
+                noise_match = NOISE_PATTERN.match(line)
+                if noise_match:
+                    output.append(noise_match.group(1))
+                else:
+                    # Something is wrong so empty output is returned.
+                    pass
```

Comment:
I think there are some code duplication here. In addition, maybe you could refactor it to reduce the indentation layers.

Reply:
No code duplication. Notice that different matchers are used. I did reduce the indentation.